src/autopas/selectors/tuningStrategy/TuningStrategySuperClass.h

```diff
@@ -0,0 +1,130 @@
+   * @param allowedContainerOptions
+   * @param allowedTraversalOptions
+   * @param allowedDataLayoutOptions
+   * @param allowedNewton3Options
+   * @param allowedCellSizeFactors
+   */
+  TuningStrategySuperClass(const std::set<ContainerOption> &allowedContainerOptions,
+                           const std::set<double> &allowedCellSizeFactors,
+                           const std::set<TraversalOption> &allowedTraversalOptions,
+                           const std::set<DataLayoutOption> &allowedDataLayoutOptions,
+                           const std::set<Newton3Option> &allowedNewton3Options)
+      : _containerOptions(allowedContainerOptions) {
+    // sets search space and current config
+    populateSearchSpace(allowedContainerOptions, allowedCellSizeFactors, allowedTraversalOptions,
+                        allowedDataLayoutOptions, allowedNewton3Options);
+  }
+
+  /**
+   * Constructor for the TuningStrategySuperClass that only contains the given configurations.
+   * This constructor assumes only valid configurations are passed! Mainly for easier unit testing.
+   * @param allowedConfigurations Set of configurations AutoPas can choose from.
+   */
+  explicit TuningStrategySuperClass(std::set<Configuration> allowedConfigurations)
+      : _containerOptions{}, _searchSpace(std::move(allowedConfigurations)) {
+    for (const auto &config : _searchSpace) {
+      _containerOptions.insert(config.container);
+    }
+  }
+
+  inline std::set<ContainerOption> getAllowedContainerOptions() const override { return _containerOptions; }
+
+  inline bool searchSpaceIsTrivial() const override { return _searchSpace.size() == 1; }
+
+  inline bool searchSpaceIsEmpty() const override { return _searchSpace.empty(); }
+
+ protected:
+  /**
+   * Fills the search space with the cartesian product of the given options (minus invalid combinations).
+   * @param allowedContainerOptions
+   * @param allowedCellSizeFactors
+   * @param allowedTraversalOptions
+   * @param allowedDataLayoutOptions
+   * @param allowedNewton3Options
+   */
+  inline void populateSearchSpace(const std::set<ContainerOption> &allowedContainerOptions,
+                                  const std::set<double> &allowedCellSizeFactors,
+                                  const std::set<TraversalOption> &allowedTraversalOptions,
+                                  const std::set<DataLayoutOption> &allowedDataLayoutOptions,
+                                  const std::set<Newton3Option> &allowedNewton3Options);
+
+  /**
+   * Finds the optimum in an unordered map
+   * @param findOptimumSearchSpace
+   * @return optimum
+   */
+  static inline std::__detail::_Node_const_iterator<std::pair<const Configuration, unsigned long>, false, true>
+  getOptimum(const std::unordered_map<Configuration, size_t, ConfigHash> &findOptimumSearchSpace);
+
+  /**
+   * Contains every container option.
```

Comment:
```suggestion
   * Contains every allowed container option.
```
maybe also rename the member?